packages/storage-plus/src/indexed_snapshot.rs

```diff
@@ -212,12 +213,90 @@ where
     }
 }
 
+#[cfg(feature = "iterator")]
+impl<'a, K, T, I> IndexedSnapshotMap<'a, K, T, I>
+where
+    T: Serialize + DeserializeOwned + Clone,
+    K: PrimaryKey<'a>,
+    I: IndexList<T>,
+{
+    pub fn sub_prefix_de(&self, p: K::SubPrefix) -> Prefix<K::SuperSuffix, T> {
+        Prefix::new(self.pk_namespace, &p.prefix())
+    }
+
+    pub fn prefix_de(&self, p: K::Prefix) -> Prefix<K::Suffix, T> {
+        Prefix::new(self.pk_namespace, &p.prefix())
+    }
+}
+
+#[cfg(feature = "iterator")]
+impl<'a, K, T, I> IndexedSnapshotMap<'a, K, T, I>
+where
+    T: Serialize + DeserializeOwned + Clone,
+    K: PrimaryKey<'a> + KeyDeserialize,
+    I: IndexList<T>,
+{
+    /// while range_de assumes you set the prefix to one element and call range over the last one,
```

Comment:
I'd be inclined to rephrase this part, personally I'm not sure I follow:
> assumes you set the prefix to one element

meaning assumes that each element has prefix set?
and 
> call range over the last one

meaning call range over last one (element)? So since last element?

Reply:
The difference is somewhat subtle: `range` over a `prefix` only iterates over all the elements with **that** prefix. Whereas `prefix_range` allows you to set the range directly over the prefix. Meaning prefix can now vary (inclusively or exclusively, depending on `PrefixBound`) between the `min` and `max` `prefix_range` parameters, if specified.

Reply:
I'll try to clarify this in the comment.